### ISSUE TITLE
Unable to parse iperf3 output as JSON error

### ISSUE THREAD

**Issue by tonin** [perfsonar/pscheduler#306]
I have a recurring JSON parsing error on the QALAB host as can be seen on the central MA or on the host itself.  For example: https://ps-4-0.qalab.geant.net/perfsonar-graphs/?source=150.254.208.65&dest=140.182.44.177&url=https%3A%2F%2Fps-4-0.qalab.geant.net%2Fesmond%2Fperfsonar%2Farchive%2F#start=1486659317&end=1486745717&timeframe=1d&summaryWindow=300

I also see the error when running from the CLI:
```
antoine^2@ps-4-0:(~$ pscheduler task --slip PT60M throughput --duration PT20S --source ps-4-0.qalab.geant.net --ip-version 4 --dest perfsonar-dev5.grnoc.iu.edu --parallel 1
Submitting task...
Task URL:
https://ps-4-0.qalab.geant.net/pscheduler/tasks/d1e2a68a-7402-4415-b386-58dd567a7c32
Fetching first run...

Next run:
https://ps-4-0.qalab.geant.net/pscheduler/tasks/d1e2a68a-7402-4415-b386-58dd567a7c32/runs/f939f80e-7a19-4e21-bce0-91001e45090f
Starts 2017-02-13T18:29:57+01:00 (~1878 seconds)
Ends   2017-02-13T18:31:18+01:00 (~80 seconds)
Waiting for result...
Unable to retrieve formatted result: Failed to format result: Traceback (most recent call last):
  File "/usr/lib/pscheduler/commands/../classes/test/throughput/result-format", line 25, in <module>
    valid, message = validate.result_is_valid(input["result"])
  File "/usr/lib/pscheduler/classes/test/throughput/validate.py", line 177, in result_is_valid
    return json_validate(input_json, RESULT_SCHEMA)
  File "/usr/lib/python2.7/dist-packages/pscheduler/jsonval.py", line 709, in json_validate
    '/' + ('/'.join([str(x) for x in ex.absolute_path])),
AttributeError: 'ValidationError' object has no attribute 'absolute_path'
```

The output of the run is visible at https://ps-4-0.qalab.geant.net/pscheduler/tasks/d1e2a68a-7402-4415-b386-58dd567a7c32/runs/f939f80e-7a19-4e21-bce0-91001e45090f

The iperf3 version deployed on the QALAB host is 3.1.6. Is there anything more I can do to debug this JSON parsing issue?

**Comment by tonin** [comment]
After a pscheduler debug session, I see that the command called is `/usr/bin/bwctl -v -T iperf3 -s ps-4-0.qalab.geant.net -c perfsonar-dev5.grnoc.iu.edu -t 20 --parsable -P 1 -4`

When I run that command from the CLI, I have:

```
antoine^2@ps-4-0:(~$ /usr/bin/bwctl -v -T iperf3 -s ps-4-0.qalab.geant.net -c perfsonar-dev5.grnoc.iu.edu -t 20 --parsable -P 1 -4
Messages being sent to syslog(user,err)
bwctl: Using ps-4-0.qalab.geant.net as the address for remote sender
bwctl: Using perfsonar-dev5.grnoc.iu.edu as the address for remote receiver
bwctl: Available in-common: iperf nuttcp iperf3 
bwctl: Using tool: iperf3
bwctl: Server 'perfsonar-dev5.grnoc.iu.edu' accepted test request at time 1487022327.789913
bwctl: Client 'ps-4-0.qalab.geant.net' accepted test request at time 1487022327.789913
bwctl: 25 seconds until test results available

SENDER START
bwctl: start_endpoint: 3696011123.114403
bwctl: run_endpoint: receiver: 140.182.44.177
bwctl: run_endpoint: sender: 150.254.208.65
bwctl: exec_line: iperf3 -c 140.182.44.177 -B 150.254.208.65 -f m -J -p 5082 -V -Z -t 20 -P 1
bwctl: run_tool: tester: iperf3
bwctl: run_tool: receiver: 140.182.44.177
bwctl: run_tool: sender: 150.254.208.65
bwctl: start_tool: 3696011127.789972
Control connection MSS 1448
{
        "start":        {
                "connected":    [],
                "version":      "iperf 3.1.6",
                "system_info":  "Linux ps-4-0 4.4.0-59-generic #80~14.04.1-Ubuntu SMP Fri Jan 6 18:02:02 UTC 2017 x86_64"
        },
        "intervals":    [{
                        "streams":      []
                }],
        "end":  {
                "streams":      [],
                "sum_sent":     {
                        "start":        0,
                        "end":  0,
                        "seconds":      0,
                        "bytes":        0,
                        "bits_per_second":      0,
                        "retransmits":  0
                },
                "sum_received": {
                        "start":        0,
                        "end":  0,
                        "seconds":      0,
                        "bytes":        0,
                        "bits_per_second":      0
                },
                "cpu_utilization_percent":      {
                        "host_total":   0.019166,
                        "host_user":    0,
                        "host_system":  0,
                        "remote_total": 0,
                        "remote_user":  0,
                        "remote_system":        0
                }
        },
        "error":        "interrupt - the client has terminated"
}

bwctl: remote peer cancelled test
bwctl: stop_tool: 3696011128.054627
bwctl: stop_endpoint: 3696011128.054627

SENDER END
```

**Comment by mfeit-internet2** [comment]
Well, that rules out pScheduler.  ;-)

The last problem that was patched in iperf3 was a verbose message that was printed no matter what output mode the program was in.   This may be the same thing.

**Comment by tonin** [comment]
Should be solved by dc445fa6c8fb52279aae76c90dcaf0e3c4f2c824
